src/components/tableview/TableTable.vue

```diff
@@ -484,51 +519,135 @@ export default {
+    },
+    addPendingChange(changeType, payload) {
+      if (changeType === CHANGE_TYPE_UPDATE) {
+        // remove existing pending updates with identical pKey-column combo
+        let pendingUpdates = _.reject(this.pendingChanges.updates, { 'key': payload.key })
+        pendingUpdates.push(payload)
+
+        this.$set(this.pendingChanges, 'updates', pendingUpdates)
+      }
+
+      if (changeType === CHANGE_TYPE_DELETE) {
+        // remove pending updates for the row marked for deletion
+        let filter = { 'primaryKey': payload.primaryKey }
+        let discardedUpdates = _.filter(this.pendingChanges.updates, filter)
+        let pendingUpdates = _.reject(this.pendingChanges.updates, filter)
+
+        discardedUpdates.forEach(update => this.discardColumnUpdate(update))
+
+        this.$set(this.pendingChanges, 'updates', pendingUpdates)
+
+        if (!_.find(this.pendingChanges.deletes, { 'primaryKey': payload.primaryKey })) {
+          this.pendingChanges.deletes.push(payload)
         }
-        log.info("new Data: ", newData)
-        this.pendingEdits = {}
-      } catch (ex) {
-        this.pendingEditList.forEach(edit => {
-          edit.cell.getElement().classList.add('edit-error')
-        })
-        this.setQueryError('Error Saving Changes', ex.message)
       }
+    },
+    resetPendingChanges() {
+      this.pendingChanges = {
+        inserts: [],
+        updates: [],
+        deletes: []
+      }
+    },
+    async saveChanges() {
+        let replaceData = false
 
+        // handle updates
+        if (this.hasPendingUpdates) {
+          try {
+            const result = await this.connection.updateValues(this.pendingChanges.updates)
+            const updateIncludedPK = this.pendingChanges.updates.find(e => e.column === e.pkColumn)
+
+            if (updateIncludedPK) {
+              replaceData = true
+            } else {
+              this.tabulator.updateData(result)
+              this.pendingChanges.updates.forEach(edit => {
+                edit.cell.getElement().classList.remove('edited')
+                edit.cell.getElement().classList.add('edit-success')
+                setTimeout(() => {
+                  if (edit.cell.getElement()) {
+                    edit.cell.getElement().classList.remove('edit-success')
+                  }
+                }, 1000)
+              })
+            }
+
+            this.pendingChanges.updates = []
+            log.info("new Data: ", result)
+          } catch (ex) {
+            this.pendingChanges.updates.forEach(edit => {
+              edit.cell.getElement().classList.add('edit-error')
+            })
+            this.setQueryError('Error Saving Changes', ex.message)
+            this.$noty.error("Error updating rows")
+
+            return
+          }
+        }
+
+        // handle deletes
+        if (this.hasPendingDeletes) {
+          try {
+            await this.connection.deleteRows(this.pendingChanges.deletes)
```

Comment:
I think both the updates, and the deletes should happen inside a single transaction. We don't really want changes to be partially applied.